src/translate.directive.ts

```diff
@@ -27,6 +28,13 @@ export class TranslateDirective implements AfterViewChecked, OnDestroy {
                 this.checkNodes(event.translations);
             });
         }
+
+        // subscribe to onDefaultLangChange event, in case the default language changes
+        if(!this.onDefaultLangChangeSub) {
+            this.onDefaultLangChangeSub = this.translateService.onDefaultLangChange.subscribe((event: DefaultLangChangeEvent) => {
+                this.checkNodes(event.translations);
```

Comment:
You are right that changing the default lang is not changing the current lang, but it could potentially change any of the values right?
I mean, we could implement something which only checks the translations that are not translated yet (displaying the translation's key). That would make it more efficient, but would also require some more effort.